File: src/main/java/no/nav/sosialhjelp/soknad/domain/model/digisosapi/FilOpplasting.java
package no.nav.sosialhjelp.soknad.domain.model.digisosapi;

import java.io.InputStream;


public class FilOpplasting {

    public FilMetadata metadata;
    public InputStream data;

    public FilOpplasting(FilMetadata metadata, InputStream data) {
        this.metadata = metadata;
        this.data = data;
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/innsending/digisosapi/dto/FilOpplasting.kt
package no.nav.sosialhjelp.soknad.innsending.digisosapi.dto

import java.io.InputStream

data class FilOpplasting(
    val metadata: FilMetadata,
    val data: InputStream
)

data class FilMetadata(
    val filnavn: String,
    val mimetype: String,
    val storrelse: Long
)

data class FilForOpplasting<T>(
    val filnavn: String,
    val metadata: T,
    val data: InputStream
)
